tmt/steps/provision/local.py

```diff
@@ -133,6 +133,43 @@ def pull(
             extend_options: Optional[list[str]] = None) -> None:
         """ Nothing to be done to pull workdir """
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """fetch and return content of a requested log"""
+        if log_name == 'dmesg':
+            return self.execute(Command('dmesg')).stdout
+        return None
+
+    def store_log(
+            self,
+            log_name: Optional[str],
+            log_path: Optional[Path],
```

Comment:
1. As we discussed, `log_path` is probably not something optional, it's a given input parameter. I can imagine that it might be a directory, then the method would save the log into this directory using `log_name`, but still, I see no use for `log_path` being unset. That would mean the guest implementation would need to decide the destination, and that's something we should avoid as it lacks context.
2. `log_name`, being optional, deserves a default value, i.e. `log_name: Optional[str] = None`.

Reply:
> 1. As we discussed, `log_path` is probably not something optional, it's a given input parameter. I can imagine that it might be a directory, then the method would save the log into this directory using `log_name`, but still, I see no use for `log_path` being unset. That would mean the guest implementation would need to decide the destination, and that's something we should avoid as it lacks context.

Got it, I was thinking maybe users want to store the logs to dir they ran tmt command:)

> 
>     2. `log_name`, being optional, deserves a default value, i.e. `log_name: Optional[str] = None`.

Yeah,will updated accordingly

Reply:
> Got it, I was thinking maybe users want to store the logs to dir they ran tmt command:)

Maybe, that sounds like something users might do, but then again, it would not be something for `store_log()` to decide, but for other parts of tmt, e.g `tmt.base.Run` which implements `tmt run`: this class would decide where to put logs, and it would task `provision` step with the right directory.